src/peergos/server/tests/SpaceCheckingKeyFilterTests.java

```diff
@@ -0,0 +1,46 @@
+package peergos.server.tests;
+import org.junit.*;
+import static org.junit.Assert.*;
+import peergos.server.SpaceCheckingKeyFilter;
+import peergos.shared.Crypto;
+import peergos.shared.cbor.CborObject;
+import peergos.shared.crypto.hash.PublicKeyHash;
+import peergos.shared.crypto.hash.Sha256;
+import peergos.shared.io.ipfs.multihash.Multihash;
+import peergos.shared.merklebtree.MaybeMultihash;
+
+import java.util.HashSet;
+import java.util.Random;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.Map;
+import static peergos.server.SpaceCheckingKeyFilter.*;
+
+public class SpaceCheckingKeyFilterTests {
+    private static Random RANDOM = new Random(666);
+    private final Crypto crypto = Crypto.initJava();
+
+    private static byte[] random() {
+        byte[] bytes = new byte[Multihash.Type.sha2_256.length];
+        RANDOM.nextBytes(bytes);
+        return bytes;
+    }
+    @Test
+    public void serdeTest() {
```

Comment:
what is serde?

Reply:
serialize-deserialize

Reply:
Not obvious to me

Reply:
It's a thing :)

https://github.com/serde-rs/serde

I'm bringing it back